doc/architecture.rst

```diff
@@ -526,20 +527,22 @@ See `Set Pool Values`_ for details.
 Mapping PGs to OSDs
 ~~~~~~~~~~~~~~~~~~~
 
-Each pool has a number of placement groups. CRUSH maps PGs to OSDs dynamically.
-When a Ceph Client stores objects, CRUSH will map each object to a placement
-group.
-
-Mapping objects to placement groups creates a layer of indirection between the
-Ceph OSD Daemon and the Ceph Client. The Ceph Storage Cluster must be able to
-grow (or shrink) and rebalance where it stores objects dynamically. If the Ceph
-Client "knew" which Ceph OSD Daemon had which object, that would create a tight
-coupling between the Ceph Client and the Ceph OSD Daemon. Instead, the CRUSH
-algorithm maps each object to a placement group and then maps each placement
-group to one or more Ceph OSD Daemons. This layer of indirection allows Ceph to
-rebalance dynamically when new Ceph OSD Daemons and the underlying OSD devices
-come online. The following diagram depicts how CRUSH maps objects to placement
-groups, and placement groups to OSDs.
+Each pool has a number of placement groups (PGs) within it. CRUSH dynamically
+maps PGs to OSDs. When a Ceph Client stores objects, CRUSH maps each object
+to a PG. 
+
+Mapping objects to PGs groups creates a "layer of indirection" between the Ceph
+OSD Daemon and the Ceph Client. The Ceph Storage Cluster must be able to grow
+(or shrink) and rebalance places where it stores objects dynamically. 
+
+If the Ceph Client "knew" which Ceph OSD Daemons were storing which objects, a
+tight coupling would exist between the Ceph Client and the Ceph OSD Daemon.
+But Ceph avoids any such tight coupling. Instead, the CRUSH algorithm maps each
+object to a placement group and then maps each placement group to one or more
+Ceph OSD Daemons. This "layer of indirection" allows Ceph to rebalance
```

Comment:
This kinda duplicates the above lines.

Reply:
I feel that way too, but I'm going to leave this in for now. A much later pass over this document might remove this repetition, but I want to keep it in so that the readers encounter the idea twice in rapid succession. I also have a strong desire to call the "layer of indirection" a "layer of abstraction".